lib/rules/declaration-block-no-redundant-longhand-properties/index.mjs

```diff
@@ -272,12 +278,52 @@ const rule = (primary, secondaryOptions) => {
 				const prop = decl.prop.toLowerCase();
 				const unprefixedProp = vendor.unprefixed(prop);
 				const prefix = vendor.prefix(prop);
+
+				// Check if this is a shorthand property and track it
+				const shorthandLonghands = longhandSubPropertiesOfShorthandProperties.get(
+					/** @type {import('stylelint').ShorthandProperties} */ (unprefixedProp),
+				);
+
+				if (shorthandLonghands) {
+					// This is a shorthand property, store it with its value
+					declaredShorthands.set(prop, decl.value.trim());
```

Comment:
[suggestion] Can you reduce repeated `decl.value.trim()` by adding a new local variable?

```js
eachDecl((decl) => {
  const declValue = decl.value.trim();
```

```suggestion
					declaredShorthands.set(prop, declValue);
```